src/OAuth2Form.tsx

```diff
@@ -130,10 +132,19 @@ export default class OAuth2Form
     }
     
     // if user has already consented to this client request, no need to ask again
-    getHasAlreadyConsented(token:string) {
+    async getHasAlreadyConsented(token:string) {
         if (token && !this.state.error) {
             const code = getURLParam('code')
             if (code) return; // we're in the middle of a SSO, do not attempt to get OAuth2RequestDescription yet
+            const maxAgeURLParam = getURLParam('max_age')
+            // SWC-5597: if max_age is defined, then return if the user last authenticated more than max_age seconds ago
+            if (maxAgeURLParam && parseInt(maxAgeURLParam)) {
```

Comment:
note that if [parseInt()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) is given an invalid number, it will return NaN (which is falsy, so it will not respect the parameter).